src/tsd/HttpQuery.java

```diff
@@ -64,7 +57,7 @@
  * It makes it easier to provide a few utility methods to respond to the
  * requests.
  */
-final class HttpQuery {
+final class HttpQuery extends AbstractHttpQuery {
```

Comment:
There are many more functions in here that can move into the base class. You could really just leave the built-in homepage bits in here and move everything else. Would that work for you?